### PR TITLE
Themes: Remove redundant actions and reducers

### PR DESCRIPTION
**Pull request by seear**
No visual changes. This PR removes redundant code.

It is not necessary to have both THEMES_RECEIVE and THEMES_REQUEST_SUCCESS
action types, since they both signify an array of themes being returned
from a server.

Remove THEMES_RECEIVE and allow the receiveThemes() action to accept
a query, allowing it to take the place of receiveThemesQuery().

**To Test**
Go to http://calypso.localhost:3000/design
* Check that theme searches and queries work exactly as before, for jetpack and non-jetpack sites
* Check that activating a theme works as before
* Check that the _CURRENT THEME_ bar at the top of the theme showcase works as before

### CODE REVIEW

**Comment by seear** [comment]
Now that we only have one main `receive...` action, we no longer need the separate [`filterThemes()`](https://github.com/Automattic/wp-calypso/pull/11782/files#diff-dd8c5284a85f9baf309a0ec3a482d228R132), or even the `shouldFilterWpcomThemes()` selector, so I looked extracting them back into `receiveThemes()`.

In the end I decided that having them as separate functions aids readability. Happy to reconsider if anyone feels otherwise.

**Comment by seear** [comment]
> In the end I decided that having them as separate functions aids readability

reconsidered on `filterThemes` after #11787.

File: client/state/ui/action-log/reducer.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const relevantTypes = {
 	EDITOR_PASTE_EVENT,
 	FIRST_VIEW_HIDE,
 	GUIDED_TOUR_UPDATE,
-	THEMES_RECEIVE,
+	THEMES_REQUEST_SUCCESS,
```
Comment (seear) [inline client/state/ui/action-log/reducer.js]: @mcsf is this an appropriate change now that 'THEMES_REQUEST_SUCCESS` is the sole signifier of themes arriving at the client?

File: client/state/themes/actions.js [hunk 2]
```diff
@@ -95,87 +94,49 @@ export function receiveTheme( theme, siteId ) {
...
+			found = filteredThemes.length;
+		}
 
 		dispatch( {
 			type: THEMES_REQUEST_SUCCESS,
 			themes: filteredThemes,
 			siteId,
 			query,
-			found: found,
+			found,
```
Comment (seear) [inline client/state/themes/actions.js]: Can't decide if we should continue to allow `found` to be `undefined` here or not.

**Comment by ockham** [comment]
Getting a wrong uploaded theme count of 1 on JP site now:

File: client/state/ui/action-log/reducer.js (hunk 1 above)
Comment (mcsf) [inline client/state/ui/action-log/reducer.js]: Yep, should be fine! :) 👌

**Comment by ockham** [comment]
Rebased.

File: client/state/themes/actions.js [hunk 3]
```diff
@@ -94,87 +93,49 @@ export function receiveTheme( theme, siteId ) {
...
-			themes,
-			siteId,
-			filterWpcom,
-			query,
-			foundCount,
-		);
+		let filteredThemes = themes;
+		let found = foundCount;
+
+		if ( siteId !== 'wporg' && siteId !== 'wpcom' ) {
```
Comment (ockham) [inline client/state/themes/actions.js]: Now that it's inlined, maybe we should turn this check into an 

```js
if( isJetpackSite( getState(), siteId )
```

? That'd be a bit more semantic -- theoretically, `siteId` could be a numeric WP.com `siteId`, in which case we wouldn't want to filter. (Right now, we're relying on consumers to not pass a WP.com `siteId` here, but better safe than sorry, I guess...)

File: client/state/themes/reducer.js [hunk 4]
```diff
@@ -309,7 +309,14 @@ export const queries = ( () => {
...
 		[ THEMES_REQUEST_SUCCESS ]: ( state, { siteId, query, themes, found } ) => {
-			return applyToManager( state, siteId, 'receive', true, themes, { query, found } );
+			return applyToManager(
+				state,
+				siteId,
+				'receive',
+				true,
+				themes,
+				query ? { query, found } : {}
+			);
```
Comment (ockham) [inline client/state/themes/reducer.js]: Is it possible that we don't actually need this change? I think [this](https://github.com/Automattic/wp-calypso/pull/11782/commits/6a6461a31c04887038cfc57caa669000d5798b1d#diff-dd8c5284a85f9baf309a0ec3a482d228R105) is enough.

**Comment by ockham** [comment]
Couple of nits aiming to keep things as lean as possible 😄 Feel free to 🚢 after addressing.

**Pull request merged**